src/Events/Integrations/Modules/Tickets_Emails/Email/Ticket.php

```diff
@@ -0,0 +1,161 @@
+
+use TEC\Events\Integrations\Modules\Tickets_Emails\Emails as TEC_Email_Handler;
+use TEC\Tickets\Emails\Email\Ticket as Ticket_Email;
+use TEC\Tickets\Emails\Email_Abstract;
+use Tribe\Events\Views\V2\iCalendar\Links\Google_Calendar;
+use TEC\Events\Integrations\Modules\Tickets_Emails\Template;
+
+/**
+ * Class Ticket.
+ *
+ * @since TBD
+ *
+ * @package TEC\Events\Integrations\Plugins\Event_Tickets
+ */
+class Ticket {
+	/**
+	 * The option key for the Event calendar links.
+	 *
+	 * @see Email_Abstract::get_option_key() for option key format.
+	 *
+	 * @since TBD
+	 *
+	 * @var string
+	 */
+	public static $option_add_event_links = 'tec-tickets-emails-ticket-add-event-links';
+
+	/**
+	 * The option key for the Event calendar invite.
+	 *
+	 * @see Email_Abstract::get_option_key() for option key format.
+	 *
+	 * @since TBD
+	 *
+	 * @var string
+	 */
+	public static $option_add_event_ics = 'tec-tickets-emails-ticket-add-event-ics';
+
+	/**
+	 * Filter the email settings and add TEC specific settings.
+	 *
+	 * @since TBD
+	 *
+	 * @param array<array<string,mixed>> $settings The email settings.
+	 *
+	 * @return array<array<string,mixed>> $settings The modified email settings.
+	 */
+	public function include_settings( $settings ): array {
+
+		$settings[ self::$option_add_event_links ] = [
+			'type'            => 'toggle',
+			'label'           => esc_html__( 'Include "Add to calendar" links', 'the-events-calendar' ),
+			'tooltip'         => esc_html__( "Include links to add the event to the user's calendar.", 'the-events-calendar' ),
+			'default'         => true,
+			'validation_type' => 'boolean',
+		];
+
+		$settings[ self::$option_add_event_ics ] = [
+			'type'            => 'toggle',
+			'label'           => esc_html__( 'Attach calendar invites', 'the-events-calendar' ),
+			'tooltip'         => esc_html__( 'Attach calendar invites (.ics) to the ticket email.', 'the-events-calendar' ),
+			'default'         => true,
+			'validation_type' => 'boolean',
+		];
+
+		return $settings;
+	}
+
+	/**
+	 * Filters the attachments for the Tickets Emails and maybe add the calendar ics file.
+	 *
+	 * @since TBD
+	 *
+	 * @param array<string,string> $attachments The placeholders for the Tickets Emails.
+	 * @param string               $email_id     The email ID.
+	 * @param Email_Abstract       $email_class  The email class.
+	 *
+	 * @return array<string,string> The filtered attachments for the Tickets Emails.
+	 */
+	public function include_attachments( $attachments, $email_id, $email_class ) {
+		if ( ! $email_class->is_enabled() ) {
+			return $attachments;
+		}
+
+		if ( ! tribe_is_truthy( tribe_get_option( self::$option_add_event_ics, true ) ) ) {
+			return $attachments;
+		}
+
+		$post_id = $email_class->__get( 'post_id' );
+
+		if ( ! tribe_is_event( $post_id ) ) {
+			return $attachments;
+		}
+
+		$event = tribe_get_event( $post_id );
+
+		if ( empty( $event ) ) {
+			return $attachments;
+		}
```

Comment:
This seems redundant. I think all we need is the last 5 lines here:
```suggestion
		$event = tribe_get_event( $post_id );

		if ( empty( $event ) ) {
			return $attachments;
		}
```

Reply:
If you use `tribe_get_event` on a post it'll return the post decorated with some nulls, it's not checking if it's an event. 

Posts/Pages and other CPTs won't have a calendar invite. We need both conditionals.

Reply:
So where are you specifically checking to see if it is an event within these lines? It seems that you are only checking to see if it is `null` twice.

Reply:
Okay. Just put my glasses on. That first call is `tribe_is_event` not `tribe_get_event`. My apologies. Disregard. Nothing to see here...